tests/test_utils.py

```diff
@@ -507,161 +506,156 @@ def test_gather_dimensions_exceptions():
             None,
             "blue",
-            utils.ERROR_SPECIFIED_NONPRESENT_REDUCE_DIMENSION2,
+            utils.ERROR_SPECIFIED_NONPRESENT_REDUCE_DIMENSION,
         ),
     ],
 )
-def test_gather_dimensions2_exceptions(
-    fcst, obs, weights, reduce_dims, preserve_dims, special_fcst_dims, error_msg_snippet
+def test_gather_dimensions_score_specific_fcst_dims_exceptions(
+    fcst_dims, obs_dims, weights_dims, reduce_dims, preserve_dims, score_specific_fcst_dims, error_msg_snippet
 ):
     """
-    Confirm `gather_dimensions2` raises exceptions as expected.
+    Confirm `gather_dimensions` raises exceptions as expected.
     """
+
+    assert score_specific_fcst_dims is not None
+
     with pytest.raises(ValueError) as excinfo:
-        gather_dimensions2(
-            fcst,
-            obs,
-            weights=weights,
+        gd(
+            fcst_dims,
+            obs_dims,
+            weights_dims=weights_dims,
             reduce_dims=reduce_dims,
             preserve_dims=preserve_dims,
-            special_fcst_dims=special_fcst_dims,
+            score_specific_fcst_dims=score_specific_fcst_dims,
         )
-    assert error_msg_snippet in str(excinfo.value)
-
 
-def test_gather_dimensions2_warnings():
-    """Tests that gather_dimensions2 warns as expected with correct output."""
-    # Preserve "all" as a string but named dimension present in data
-    with pytest.warns(UserWarning):
-        result = gather_dimensions2(
-            utils_test_data.DA_R.rename({"red": "all"}), utils_test_data.DA_R, preserve_dims="all"
-        )
-        assert result == set([])
-
-    with pytest.warns(UserWarning):
-        result = gather_dimensions2(
-            utils_test_data.DA_R.rename({"red": "all"}), utils_test_data.DA_R, reduce_dims="all"
-        )
-        assert result == {"red", "all"}
+    assert error_msg_snippet in str(excinfo.value)
 
 
 @pytest.mark.parametrize(
-    ("fcst", "obs", "weights", "reduce_dims", "preserve_dims", "special_fcst_dims", "expected"),
+    ("fcst_dims", "obs_dims", "weights_dims", "reduce_dims", "preserve_dims", "score_specific_fcst_dims", "expected"),
     [
-        # test that fcst and obs dims are returned
-        (utils_test_data.DA_B, utils_test_data.DA_R, None, None, None, None, {"blue", "red"}),
-        # test that fcst, obs and weights dims are returned
-        (utils_test_data.DA_B, utils_test_data.DA_R, utils_test_data.DA_G, None, None, None, {"blue", "red", "green"}),
-        # two tests that fcst, obs and weights dims are returned, without the special fcst dim
-        (utils_test_data.DA_B, utils_test_data.DA_R, utils_test_data.DA_G, None, None, "blue", {"red", "green"}),
-        (utils_test_data.DA_B, utils_test_data.DA_R, utils_test_data.DA_G, None, None, ["blue"], {"red", "green"}),
-        # test that reduce_dims="all" behaves as expected
-        (utils_test_data.DA_B, utils_test_data.DA_R, utils_test_data.DA_G, "all", None, None, {"blue", "red", "green"}),
```

Comment:
This test case needs to be moved across as it's missing

Reply:
I believe this is done now, in a new test function. Please check it I got the correct cases.

Reply:
Yes that looks good. You still need to run black